File: problems/Reverse Words in a String/reverseWords.cs
public class Solution {
    public string ReverseWords(string s) {
        var store = s.Split(" ");
        var stack = new Stack<string>();

        foreach (var item in store) {
            if (!String.IsNullOrEmpty(item)) {
                stack.Push(item);
            }
        }

        return String.Join(" ", stack);
    }
}

File: problems/Reverse Words in a String/reverseWords.java
class Solution {
    public String reverseWords(String s) {
        Stack<String> stack = new Stack<>();
        StringBuilder sb = new StringBuilder();

        for (String item : s.split(" ")) {
            if (null != item && !"".equals(item)) {
                stack.push(item);
            }
        }

        while (!stack.isEmpty()) {
            if (0 != sb.length()) {
                sb.append(" ");
            }
            sb.append(stack.pop());
        }

        return sb.toString();
    }
}
